op-e2e/actions/l2_batcher_test.go

```diff
@@ -87,3 +89,108 @@ func TestBatcher(gt *testing.T) {
 	require.False(t, isPending)
 	require.NotNil(t, vTx)
 }
+
+func TestL2Finalization(gt *testing.T) {
```

Comment:
I see that this test depends upon some of the changes in this PR (type changes + won't run if I just bring the test to develop), but it does seem like a grab bag of changes to have `op-e2e` driver fixes, a new test, & `op-node` changes bundled into one.

Reply:
The proposer changes required a more consistent view of the chain, with accurate view handling bad finalization and reorgs between requests. So I split that up in a separate PR + wrote testing for it. It's not pretty, but small enough of a PR + includes testing.